consensus/wal_test.go

```diff
@@ -16,6 +19,46 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestWALTruncate(t *testing.T) {
+	walDir, err := ioutil.TempDir("", "wal")
+	if err != nil {
+		panic(fmt.Errorf("failed to create temp WAL file: %v", err))
+	}
+
+	walFile := filepath.Join(walDir, "wal")
+
+	wal, err := NewWAL(walFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	wal.group.SetHeadSizeLimit(4096) //cause fragment
+	wal.Start()
+
+	err = WALGenerateNBlocks(wal.group, 60) //60 block's size nearly 70K
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	time.Sleep(5 * time.Second) //wait groupCheckDuration, make sure RotateFile run
```

Comment:
We're trying to avoid long sleep calls in our tests. Do you think we can do something with this call? Maybe make groupCheckDuration a configuration variable

```go
# libs/autofile/group.go
type Config struct {
    GroupCheckDuration time.Duration
}

func OpenGroup(headPath string, cfg Config) (g *Group, err error) {

# consensus/wal.go
type Config struct {
    GroupConfig
}

func NewWAL(walFile string, cfg Config) (*baseWAL, error) {
```

Reply:
Not only the GroupCheckDuration, but HeadSizeLimit also need to config, this config filed seems only need changed by test. I added some member func to Group, left NewWAL concise.